### DISCUSSION TITLE
CRA v5, msw and electron.net

### DISCUSSION THREAD

**Discussion by guilima** [mswjs/msw#1043] [Q&A] [answered]
Hello guys, after updating CRA v4.0.3 to v5.0.0 dependency inside an electron project it started to thrown the error message `'Failed to execute ``setupWorker`` in a non-browser environment. Consider using ``setupServer`` for Node.js environment instead.'` on development environment. So I debugged changing between CRA version and could find into `isNodeProcess()` the `process` being read as a node application on CRA v5. Do you have any suggestion on how I could address this problem?

**Comment by kettanaito** [comment, marked as answer]
Hey, @guilima.

Could you please include the code where you're calling `worker.start()`? It may be that you do that in the wrong context. Electron comes with two processes: main (which is Node.js) and renderer (which is browser). It's important to start the worker in the letter process.

**Comment by guilima** [comment]
Fair, probably after updating CRA to v5 the spa application that are running inside electron.net started on main process for some reason. I am still not sure why that happened, but already did a test creating an electronjs app within CRA v5 and it worked like a charm. I need to understand more about how electron.net implements electron, anyway thanks!

**Reply by kettanaito** [reply]
I'm glad you've found the problematic area!
